lib/array_intersection.rb

```diff
@@ -1,4 +1,12 @@
 def intersection(list1, list2)
-  raise NotImplementedError, "Intersection not implemented"
+  intersect = Hash[*list1.collect{ |a| [a, 0] }.flatten] # hash first list
```

Comment:
You could simplify this as intersect = Hash[list1.collect{ |a| [a, 0]}] and drop the splat and flatten.